File: app/src/main/java/com/ditronic/securezipnotes/util/OnThrottleClickListener.java
package com.ditronic.securezipnotes.util;

import android.os.SystemClock;
import android.view.View;

public abstract class OnThrottleClickListener implements View.OnClickListener {

    private static final long THROTTLE_TIME = 500; // milliseconds

    private long lastClickTime = 0;
    @SuppressWarnings("unused")
    protected abstract void onThrottleClick(View v);

    @Override
    public void onClick(View v) {
        final long clickTime = SystemClock.elapsedRealtime();
        if (clickTime - lastClickTime < THROTTLE_TIME) {
            return;
        }
        lastClickTime = clickTime;
        onThrottleClick(v);
    }
}

File: app/src/main/java/com/ditronic/securezipnotes/util/OnThrottleClickListener.kt
package com.ditronic.securezipnotes.util

import android.os.SystemClock
import android.view.View

abstract class OnThrottleClickListener : View.OnClickListener {

    private var lastClickTime: Long = 0
    protected abstract fun onThrottleClick(v: View)

    override fun onClick(v: View) {
        val clickTime = SystemClock.elapsedRealtime()
        if (clickTime - lastClickTime < THROTTLE_TIME) {
            return
        }
        lastClickTime = clickTime
        onThrottleClick(v)
    }

    companion object {

        private const val THROTTLE_TIME: Long = 500 // milliseconds
    }
}
